tutorials/hyperspectral/Advanced_Spectral_Signatures.Rmd

```diff
@@ -0,0 +1,291 @@
+lower and upper bound of each of those atmopheric absopbtion bands is specified in the HDF5 file. 
+Let's read those bands and plot rectangles where the reflectance measurements are obscured by 
+atmospheric absorbtion. 
+
+```{r mask-atmospheric-absorbtion-bands}
+# grab Reflectance metadata (which contains absorption band limits)
+reflMetadata<- h5readAttributes(f,"/SJER/Reflectance" )
+
+ab1 = reflMetadata$Band_Window_1_Nanometers
+ab2 = reflMetadata$Band_Window_2_Nanometers
+
+## Plot spectral signatures again with rectangles showing the absorption bands
+ggplot()+
+  geom_line(data = Pixel.melt, mapping = aes(x=wavelengths, y=Reflectance, color=variable), lwd=1.5)+
+  geom_rect(mapping=aes(ymin=min(Pixel.melt$Reflectance),ymax=max(Pixel.melt$Reflectance), xmin=ab1[1], xmax=ab1[2]), color="black", fill="grey40", alpha=0.8)+
+  geom_rect(mapping=aes(ymin=min(Pixel.melt$Reflectance),ymax=max(Pixel.melt$Reflectance), xmin=ab2[1], xmax=ab2[2]), color="black", fill="grey40", alpha=0.8)+
+  scale_colour_manual(values = c("green2", "green4", "grey50","tan4","blue3"),
+                      labels = c("Field", "Tree", "Roof","Soil","Water"))+
+  labs(color = "Cover Type")+
+  ggtitle("Land cover spectral signatures")+
+  theme(plot.title = element_text(hjust = 0.5, size=20))+
+  xlab("Wavelength")
+```
+
+Now we can clearly see that the noisy sections of each spectral signature is within the atmospheric
+absorbtion bands. For our final step, let's take all reflectance values from within each absorbtion
+band and set them to `NA` to remove the noisy sections from the plot.
+
+```{r remove-absorbtion-band-reflectances}
+# Duplicate the spectral signatures into a new data.frame
+Pixel.melt.masked=Pixel.melt
+
+# Mask out all values within each of the two atmospheric absorbtion bands
+Pixel.melt.masked[Pixel.melt.masked$wavelengths>ab1[1]&Pixel.melt.masked$wavelengths<ab1[2],]$Reflectance=NA
+Pixel.melt.masked[Pixel.melt.masked$wavelengths>ab2[1]&Pixel.melt.masked$wavelengths<ab2[2],]$Reflectance=NA
+
+# Plot the masked spectral signatures
+ggplot()+
+  geom_line(data = Pixel.melt.masked, mapping = aes(x=wavelengths, y=Reflectance, color=variable), lwd=1.5)+
+  scale_colour_manual(values = c("green2", "green4", "grey50","tan4","blue3"),
+                      labels = c("Field", "Tree", "Roof", "Soil", "Water"))+
+  labs(color = "Cover Type")+
+  ggtitle("Land cover spectral signatures")+
+  theme(plot.title = element_text(hjust = 0.5, size=20))+
+  xlab("Wavelength")
+
+```
+
+There you have it, spectral signatures for five different land cover types, with the readings 
+from the atmospheric absorbtion bands removed.
+
+<div id="ds-challenge" markdown="1">
+### Challenge: Compare Spectral Signatures
+
+There are many interesting comparisons to make with spectral signatures. 
+Try these challenges to explore hyperspectral data further:
+
+1. Compare five different types of vegetation, and pick an appropriate color
```

Comment:
Should this be rephrased to "Compare five types of vegetation that are different from the ones you chose in the tutorial above. ..."

Reply:
I think that the user could re-use the two types of vegetation above. I more wanted to have people explore the big and small differences between the spectral signatures of different vegetation types, as opposed to the obvious differences between veg vs. soil and other comparisons already made in this tutorial.